js/adapt-contrib-bookmarking.js

```diff
@@ -139,23 +139,49 @@ define([
             this.setLocationID(menuModel.get('_id'));
         },
 
-        setupPage: function (pageView) {
-            var hasPageBookmarkObject = pageView.model.has('_bookmarking');
-            var bookmarkModel = (hasPageBookmarkObject) ? pageView.model.get('_bookmarking') : Adapt.course.get('_bookmarking');
-            this.bookmarkLevel = bookmarkModel._level;
+        /**
+         * Calculates what the bookmarking 'level' will be for any given page.
+         * First sets a default using the course-level setting (or 'component' if that's not been set)
+         * then checks to see if that's being overridden at page level or not
+         * @param {Backbone.Model} pageModel The model for the current page view
+         * @return {String} Either 'page', 'block', or 'component' - with 'component' being the default
+         */
+        getBookmarkLevel: function(pageModel) {
+            var defaultLevel = Adapt.course.get('_bookmarking')._level || 'component';
+
+            if (pageModel.has('_bookmarking')) {
```

Comment:
`has` is redundant here.
```js
var defaultLevel = Adapt.course.get('_bookmarking')._level || 'component';
var bookmarkModel = pageModel.get('_bookmarking');
var isInherit = !bookmarkModel || (!bookmarkModel._level || bookmarkModel._level === 'inherit');
if (isInherit) return defaultLevel;
return bookmarkModel._level;
```

Reply:
ta, I had a feeling there was a neater way of doing this but couldn't quite get there

Reply:
Tom Greenfield's return earlies + your named variables seem to be a good way of reducing the nesting whilst giving extra clarity. Keeping in mind to always finish on the intended behaviour of the function.